include/deal.II/particles/particle_handler.h

```diff
@@ -524,7 +524,78 @@ namespace Particles
         &data_range);
   };
 
-  /* ---------------------- inline and template functions ------------------ */
+  /**
+   * This class manages the DataOut of a Particle Handler
+   * It currently only supports witing the particle position
+   * and their ID
+   *
+   * @ingroup Particle
+   *
+   * @author : Bruno Blais, Luca Heltai 2019
+   */
+
+  template <int dim, int spacedim>
+  class ParticleOutput : public dealii::DataOutInterface<0, spacedim>
+  {
+  public:
+    ParticleOutput() = default;
+
+    void
+    build_patches(const Particles::ParticleHandler<dim, spacedim> &particles)
+    {
+      dataset_names.reserve(1);
```

Comment:
I think you want to clear all variables you're writing into at the beginning of this function, in case someone calls it twice